smolyay/grid.py

```diff
@@ -2,348 +2,429 @@
-            grid_points_basis = [b[0] for b in grid_points_basis]
-            levels = [l[0] for l in levels]
-
-        return NestedIndexGrid(grid_points_indexes, grid_points,
-                               grid_points_basis, levels)
-
-
-class TensorGridGenerator(IndexGridGenerator):
-    """Create grid points and functions for constructing a surrogate.
-
-    Depending on the dimensionality, points and basis functions provided
-    by :class:`BasisFunctionSet`, the :meth:`__call__` makes full tensor
-    grids.
-    :meth:`__call__`  generates the indexes of
-    grid points (integer grids) depending on dimensionality (full tensor grid).
-    Grid points and their functions are made by replacing the indexes with
-    corresponding point.
-    """
+    @property
+    def number_points(self):
+        """int: random seed for generating points"""
+        return self._number_points
+
+    @number_points.setter
+    def number_points(self, value):
+        if not isinstance(value, (int, numpy.random.Generator)):
+            raise TypeError("number_points must be an integer or Generator.")
+        self._number_points = value
+
+    @property
+    def method(self):
+        """str: method of sampling"""
+        return self._method
+
+    @method.setter
+    def method(self, value):
+        if value.casefold() in self.valid_methods:
+            self._method = value.casefold()
+        else:
+            raise ValueError("Method can only be latin, halton, sobol, or uniform.")
 
-    def __init__(self, basis_set):
-        super().__init__(basis_set)
+    @property
+    def seed(self):
+        """int: random seed for generating points"""
+        return self._seed
+
+    @seed.setter
+    def seed(self, value):
+        if not isinstance(value, int):
+            raise TypeError("seed must be an integer.")
+        self._seed = value
+
+    @property
+    def options(self):
+        """None or dict: additional parameters passed to QMCEngine"""
+        return self._options
 
-    def __call__(self, dimension):
-        """Make grid points and their corresponding basis functions.
+    @options.setter
+    def options(self, value):
+        self._options = value
 
-        Depending on the dimensionality, a set of grid points are generated
-        (integer grid points). Grid points of the basis function, and
-        their corresponding basis functions are then generated by replacing
-        the indexes (integer numbers) with their points and functions.
+    def _generate_points(self):
+        """Generating the Monte Carlo mulitdimensional grid set
+
+        Using the parameters of the class, generate a grid set using
+        the a Monte Carlo/Quasi Monte Carlo method.
+        """
+        self._points = self.get_random_points(
+            self.domain, self.number_points, self.method, self.seed, self.options
+        )
+
+    @staticmethod
+    def get_random_points(domain, number_points, method, seed, options=None):
```

Comment:
This is a good candidate to turn into an abstract method that returns the generator for a specific type of random point set, then you can have a common create points method that uses that generator.